pkgs/development/python-modules/ofxtools/default.nix

```diff
@@ -1,30 +1,40 @@
 { stdenv
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
+, nose
 , python
-, sqlalchemy
+, pythonOlder
 }:
 
 buildPythonPackage rec {
   pname = "ofxtools";
   version = "0.8.20";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "87245679911c0c12429a476fd269611512d3e4b44cb8871159bb76ba70f8a46f";
+  disabled = pythonOlder "3.7";
+
+  # PyPI distribution does not include tests
+  src = fetchFromGitHub {
+    owner = "csingley";
+    repo = pname;
+    rev = version;
+    sha256 = "1s3fhhmj1acnmqglh39003db0bi451m4hcrkcpyrkqf5m32lslz8";
   };
 
+  # override $HOME directory:
+  #   error: [Errno 13] Permission denied: '/homeless-shelter'
+  preCheck = ''
+    HOME=$TMPDIR
+  '';
+  checkInputs = [ nose ];
   checkPhase = ''
-    ${python.interpreter} -m unittest discover -s ofxtools
+    runHook preCheck
+    nosetests tests/*.py
+    runHook postCheck
```

Comment:
you could just do
```
  checkInputs = [ nose ];
  checkPhase = ''
    HOME=$TMPDIR nosetests tests/*.py
  '';
```
hooks aren't super useful when referencing a single package, as you are aware of all the preconditions needed to test the package

Reply:
Thanks, updated.